src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,135 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private List<Entry<K, V>>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new ArrayList[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
```

Comment:
modify getIndex method so it can handle null values instead
```suggestion
```

Reply:
So need help to fix test